charts/pomerium/templates/signing-key-secret.yaml

```diff
@@ -0,0 +1,32 @@
+{{- define "pomerium.proxy.signingKeySecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+    helm.sh/hook-delete-policy: before-hook-creation
+{{- if .Values.config.forceGenerateSigningKey }}
+    helm.sh/hook: pre-upgrade
+{{- else if .Values.config.generateSigningKey }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.proxy.signingKeySecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+{{- end -}}
```

Comment:
@tgomas is this end at the right place? there seem to be something not right with the key being generated just after

Reply:
yes it's the `end` of `pomerium.proxy.signingKeySecretObject`'s definition, which is used in both if and else blocs below. The key is only generated inside if bloc because not used in else bloc.

Reply:
ah it's not an if at the top, sorry

Reply:
While fixing key generation (it has to be b64enc 2 times) I removed the key variable declaration since it was used at only one place. Hope that's more readable